### ISSUE TITLE
`void-dom-elements-no-children` checking every .createElement and crashing

### ISSUE THREAD

**Issue by daltones** [yannickcr/eslint-plugin-react#1073]
I've just updated to v6.10.0 and enabled `void-dom-elements-no-children`.

It's crashing with this error:
```
TypeError: Cannot read property 'properties' of undefined
    at EventEmitter.CallExpression ([...]/eslint-plugin-react/lib/rules/void-dom-elements-no-children.js:121:28)
[...]
```

It happens with this simple code:
```js
document.createElement('img');
```

So here is the first problem:
- The rule thinks that every `.createElement(` is _the_ `React.createElement`.

Now testing with the actual `React`:
```js
import React from 'react';

React.createElement('img');
```
The rule will crash too.

Second problem:
- The rule doesn't treat incomplete number of `createElement`'s arguments (as while typing `React.createElement()`).

**Comment by ljharb** [comment]
The first problem is intentional, as it looks for anything that could be a createElement call first, then refines.

The second is clearly a bug.

**Comment by daltones** [comment]
Okay, but what is the point of the first problem being intentional when `document.createElement` could be a trivial false positive?

Why can't it work same way as other rules that use `pragma` setting to verify `React.(createClass|(Pure)?Component)`?

**Comment by daltones** [comment]
Oh, right, I missed something. Fixing the second problem would not warn about `document.createElement('img')`.

But maybe first problem could turn into another issue to prevent some edge cases like `objectNothingAboutReact.createElement('img', {foo: bar}, 'baz')`.

Thanks!

**Comment by ljharb** [comment]
I believe that if the bug was fixed, only React `createElement` calls would be matched.

**Comment by jpickwell** [comment]
I just installed `master` in one of my projects, and I'm no longer getting the `TypeError: Cannot read property 'properties' of undefined` error. So #1080 fixed that, which solved my problem. Thanks @jomasti.
